Add argument types to docs

```
resource "hcloud_server" "masters" {
  count       = "${var.master_count}"
  name        = "k8s02-master${count.index+1}"
  server_type = "${var.master_instance_type}"
  image       = "fedora-28"
  datacenter  = "fsn1-dc8"
  ssh_keys    = ["${hcloud_ssh_key.default.name}"]
  keep_disk   = false
  rescue      = false
  provisioner "remote-exec" {
    inline = ["sudo dnf -y install python curl salt-minion"]

    connection {
      type        = "ssh"
      user        = "root"
      private_key = "${file(var.ssh_key_private)}"
    }
  }
  provisioner "salt-masterless" {
    local_state_tree = "./salt"
    skip_bootstrap = true
    remote_state_tree = "/srv/salt"
  }
}
```
The following output shows the error I'm getting with the above Terraform file (the destroying happens because the first `terraform apply` also fails):
```
$ terraform apply terraform.tfplan
hcloud_server.masters: Destroying... (ID: 853043)
hcloud_server.masters: Destruction complete after 1s
hcloud_server.masters: Creating...
  datacenter:   "" => "fsn1-dc8"
  image:        "" => "fedora-28"
  ipv4_address: "" => "<computed>"
  ipv6_address: "" => "<computed>"
  keep_disk:    "" => "false"
  location:     "" => "<computed>"
  name:         "" => "node1"
  rescue:       "" => "0"
  server_type:  "" => "cx11"
  ssh_keys.#:   "" => "1"
  ssh_keys.0:   "" => "Terraform Deploy"
  status:       "" => "<computed>"
hcloud_server.masters: Still creating... (10s elapsed)

Error: Error applying plan:

1 error(s) occurred:

* hcloud_server.masters: 1 error(s) occurred:

* hcloud_server.masters: invalid input in field 'type' (invalid_input)

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```
Running with `TF_LOG=DEBUG` reveals that the server creation succeeds:
```
hcloud_server.masters: Creating...
  datacenter:   "" => "fsn1-dc8"
  image:        "" => "fedora-28"
  ipv4_address: "" => "<computed>"
  ipv6_address: "" => "<computed>"
  keep_disk:    "" => "false"
  location:     "" => "<computed>"
  name:         "" => "k8s02-master1"
  rescue:       "" => "0"
  server_type:  "" => "cx11"
  ssh_keys.#:   "" => "1"
  ssh_keys.0:   "" => "Terraform Deploy"
  status:       "" => "<computed>"
2018-07-07T09:56:32.404+0200 [DEBUG] plugin.terraform-provider-hcloud_v1.2.0_x4: 2018/07/07 09:56:32 [INFO] server (853070) waiting for "create_server" action to complete...
hcloud_server.masters: Still creating... (10s elapsed)
hcloud_server.masters: Still creating... (20s elapsed)
2018-07-07T09:56:48.984+0200 [DEBUG] plugin.terraform-provider-hcloud_v1.2.0_x4: 2018/07/07 09:56:48 [INFO] server (853070) "create_server" action succeeded
2018/07/07 09:56:49 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* hcloud_server.masters: invalid input in field 'type' (invalid_input)
2018/07/07 09:56:49 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* hcloud_server.masters: invalid input in field 'type' (invalid_input)
2018/07/07 09:56:49 [DEBUG] plugin: waiting for all plugin processes to complete...

Error: Error applying plan:

1 error(s) occurred:

* hcloud_server.masters: 1 error(s) occurred:

* hcloud_server.masters: invalid input in field 'type' (invalid_input)

Terraform does not automatically rollback in the face of errors.
```

* Terraform version: `Terraform v0.11.7`
* Terraform Cloud provider: `terraform-provider-hcloud_v1.2.0_x4`

Is the error in my Terraform file?

--- a/website/docs/d/floating_ip.html.md
+++ b/website/docs/d/floating_ip.html.md
@@ -30,8 +30,8 @@ data "hcloud_floating_ip" "ip_1" {
 }
 ```
 ## Argument Reference
-- `ip_address` - IP address of the Floating IP.
+- `ip_address` - (Optional, string) IP address of the Floating IP.
 
 ## Attributes Reference
-- `id` - Unique ID of the Floating IP.
-- `ip_address` - IP address of the Floating IP.
+- `id` - (int) Unique ID of the Floating IP.
+- `ip_address` - (string) IP address of the Floating IP.

--- a/website/docs/d/image.html.md
+++ b/website/docs/d/image.html.md
@@ -22,17 +22,17 @@ resource "hcloud_server" "main" {
 }
 ```
 ## Argument Reference
-- `id` - ID of the Image.
-- `name` - Name of the Image.
+- `id` - (Optional, string) ID of the Image.
+- `name` - (Optional, string) Name of the Image.
 
 ## Attributes Reference
-- `id` - Unique ID of the Image.
-- `name` - Name of the Image.
-- `type` - Type of the Image, could be `system`,`backup` or `snapshot.
-- `status` - Status of the Image.
-- `description` - Description of the Image.
-- `created` - Date when the Image was created (in ISO-8601 format).
-- `os_flavor` - Flavor of operating system contained in the image, could be `ubuntu`, `centos`, `debian`, `fedora` or `unknown`.
-- `os_version` - Operating system version.
-- `rapid_deploy` - Indicates that rapid deploy of the image is available.
-- `deprecated` - Point in time when the image is considered to be deprecated (in ISO-8601 format).
+- `id` - (int) Unique ID of the Image.
+- `name` - (string) Name of the Image.
+- `type` - (string) Type of the Image, could be `system`, `backup` or `snapshot`.
+- `status` - (string) Status of the Image.
+- `description` - (string) Description of the Image.
+- `created` - (string) Date when the Image was created (in ISO-8601 format).
+- `os_flavor` - (string) Flavor of operating system contained in the image, could be `ubuntu`, `centos`, `debian`, `fedora` or `unknown`.
+- `os_version` - (string) Operating system version.
+- `rapid_deploy` - (bool) Indicates that rapid deploy of the image is available.
+- `deprecated` - (string) Point in time when the image is considered to be deprecated (in ISO-8601 format).

--- a/website/docs/d/ssh_key.html.md
+++ b/website/docs/d/ssh_key.html.md
@@ -24,12 +24,12 @@ resource "hcloud_server" "main" {
 }
 ```
 ## Argument Reference
-- `id` - ID of the SSH Key.
-- `name` - Name of the SSH Key.
-- `fingerprint` - Fingerprint of the SSH Key.
+- `id` - (Optional, string) ID of the SSH Key.
+- `name` - (Optional, string) Name of the SSH Key.
+- `fingerprint` - (Optional, string) Fingerprint of the SSH Key.
 
 ## Attributes Reference
-- `id` - Unique ID of the SSH Key.
-- `name` - Name of the SSH Key.
-- `fingerprint` - Fingerprint of the SSH Key.
-- `public_key` - Public Key of the SSH Key.
+- `id` - (int) Unique ID of the SSH Key.
+- `name` - (string) Name of the SSH Key.
+- `fingerprint` - (string) Fingerprint of the SSH Key.
+- `public_key` - (string) Public Key of the SSH Key.

--- a/website/docs/index.html.markdown
+++ b/website/docs/index.html.markdown
@@ -34,6 +34,6 @@ resource "hcloud_server" "web" {
 
 The following arguments are supported:
 
-- `token` - (Required) This is the Hetzner Cloud API Token, can also be specified with the `HCLOUD_TOKEN` environment variable.
-- `endpoint` - Hetzner Cloud API endpoint, can be used to override the default API Endpoint `https://api.hetzner.cloud/v1`.
-- `poll_interval` -  Configures the interval in which actions are polled by the client. Default `500ms`. Increase this interval if you run into rate limiting errors. 
+- `token` - (Required, string) This is the Hetzner Cloud API Token, can also be specified with the `HCLOUD_TOKEN` environment variable.
+- `endpoint` - (Optional, string) Hetzner Cloud API endpoint, can be used to override the default API Endpoint `https://api.hetzner.cloud/v1`.
+- `poll_interval` -  (Optional, string) Configures the interval in which actions are polled by the client. Default `500ms`. Increase this interval if you run into rate limiting errors.

--- a/website/docs/r/floating_ip.html.md
+++ b/website/docs/r/floating_ip.html.md
@@ -27,21 +27,21 @@ resource "hcloud_floating_ip" "master" {
 
 ## Argument Reference
 
-- `type` - (Required) Type of the Floating IP. `ipv4` `ipv6`
-- `server_id` - (Optional) Server to assign the Floating IP to.
-- `home_location` - (Optional) Home location (routing is optimized for that location). Optional if server_id argument is passed.
-- `description` - (Optional) Description of the Floating IP.
+- `type` - (Required, string) Type of the Floating IP. `ipv4` `ipv6`
+- `server_id` - (Optional, int) Server to assign the Floating IP to.
+- `home_location` - (Optional, string) Home location (routing is optimized for that location). Optional if server_id argument is passed.
+- `description` - (Optional, string) Description of the Floating IP.
 
 ## Attributes Reference
 
-- `id` - Unique ID of the Floating IP.
-- `type` - Type of the Floating IP.
-- `server_id` - Server to assign the Floating IP is assigned to.
-- `home_location` - Home location.
-- `description` - Description of the Floating IP.
-- `ip_address` - IP Address of the Floating IP.
-- `ip_network` - IPv6 subnet. (Only set if `type` is `ipv6`)
-- `labels` - User-defined labels (key-value pairs)
+- `id` - (int) Unique ID of the Floating IP.
+- `type` - (string) Type of the Floating IP.
+- `server_id` - (int) Server to assign the Floating IP is assigned to.
+- `home_location` - (string) Home location.
+- `description` - (string) Description of the Floating IP.
+- `ip_address` - (string) IP Address of the Floating IP.
+- `ip_network` - (string) IPv6 subnet. (Only set if `type` is `ipv6`)
+- `labels` - (map) User-defined labels (key-value pairs)
 
 ## Import
 

--- a/website/docs/r/floating_ip_assignment.html.md
+++ b/website/docs/r/floating_ip_assignment.html.md
@@ -33,11 +33,11 @@ resource "hcloud_floating_ip" "master" {
 
 ## Argument Reference
 
-- `floating_ip_id` - (Required) ID of the Floating IP.
-- `server_id` - (Required) Server to assign the Floating IP to.
+- `floating_ip_id` - (Required, int) ID of the Floating IP.
+- `server_id` - (Required, int) Server to assign the Floating IP to.
 
 ## Attributes Reference
 
-- `id` - Unique ID of the Floating IP Assignment.
-- `floating_ip_id` - ID of the Floating IP.
-- `server_id` - Server the Floating IP was assigned to.
+- `id` - (int) Unique ID of the Floating IP Assignment.
+- `floating_ip_id` - (int) ID of the Floating IP.
+- `server_id` - (int) Server the Floating IP was assigned to.

--- a/website/docs/r/rdns.html.md
+++ b/website/docs/r/rdns.html.md
@@ -28,22 +28,22 @@ resource "hcloud_rdns" "master" {
 
 ## Argument Reference
 
-- `dns_ptr` - (Required) The DNS address the `ip_address` should resolve to.
-- `ip_address` - (Required) The IP address that should point to `dns_ptr`.
-- `server_id` - (Required) The server the `ip_address` belongs to.
-- `floating_ip_id` - (Required) The Floating IP the `ip_address` belongs to.
+- `dns_ptr` - (Required, string) The DNS address the `ip_address` should resolve to.
+- `ip_address` - (Required, string) The IP address that should point to `dns_ptr`.
+- `server_id` - (Required, int) The server the `ip_address` belongs to.
+- `floating_ip_id` - (Required, int) The Floating IP the `ip_address` belongs to.
 
 ## Attributes Reference
 
-- `id` - Unique ID of the Reverse DNS Entry.
-- `dns_ptr` - DNS pointer for the IP address.
-- `ip_address` - IP address.
-- `server_id` - The server the IP address belongs to.
-- `floating_ip_id` - The Floating IP the IP address belongs to.
+- `id` - (int) Unique ID of the Reverse DNS Entry.
+- `dns_ptr` - (string) DNS pointer for the IP address.
+- `ip_address` - (string) IP address.
+- `server_id` - (int) The server the IP address belongs to.
+- `floating_ip_id` - (int) The Floating IP the IP address belongs to.
 
 ## Import
 
-Reverse DNS entries can be imported using a compound ID with the following format:  
+Reverse DNS entries can be imported using a compound ID with the following format:
 `<prefix (s for server/ f for floating ip)>-<server or floating ip ID>-<IP address>`
 
 ```

--- a/website/docs/r/server.html.md
+++ b/website/docs/r/server.html.md
@@ -25,36 +25,36 @@ resource "hcloud_server" "node1" {
 
 The following arguments are supported:
 
-- `name` - (Required) Name of the server to create (must be unique per project and a valid hostname as per RFC 1123).
-- `server_type` - (Required) Name of the server type this server should be created with.
-- `image` - (Required) Name or ID of the image the server is created from.
-- `location` - (Optional) The location name to create the server in.
-- `datacenter` - (Optional) The datacenter name to create the server in.
-- `user_data` - (Optional) Cloud-Init user data to use during server creation
-- `ssh_keys` - (Optional) SSH key IDs or names which should be injected into the server at creation time
-- `keep_disk` - (Optional) If true, do not upgrade the disk. This allows downgrading the server type later.
-- `backup_window` - (Optional) Enable and configure backups for a server. Time window (UTC) in which the backup will run, choices: `22-02` `02-06` `06-10` `10-14` `14-18` `18-22`
-- `iso` - (Optional) Name of an ISO image to mount.
-- `rescue` - (Optional) Enable and boot in to the specified rescue system. This enables simple installation of custom operating systems. `linux64` `linux32` or `freebsd64`
-- `labels` - (Optional) User-defined labels (key-value pairs) should be created with.
+- `name` - (Required, string) Name of the server to create (must be unique per project and a valid hostname as per RFC 1123).
+- `server_type` - (Required, string) Name of the server type this server should be created with.
+- `image` - (Required, string) Name or ID of the image the server is created from.
+- `location` - (Optional, string) The location name to create the server in.
+- `datacenter` - (Optional, string) The datacenter name to create the server in.
+- `user_data` - (Optional, string) Cloud-Init user data to use during server creation
+- `ssh_keys` - (Optional, list) SSH key IDs or names which should be injected into the server at creation time
+- `keep_disk` - (Optional, bool) If true, do not upgrade the disk. This allows downgrading the server type later.
+- `backup_window` - (Optional, string) Enable and configure backups for a server. Time window (UTC) in which the backup will run, choices: `22-02` `02-06` `06-10` `10-14` `14-18` `18-22`
+- `iso` - (Optional, string) Name of an ISO image to mount.
+- `rescue` - (Optional, string) Enable and boot in to the specified rescue system. This enables simple installation of custom operating systems. `linux64` `linux32` or `freebsd64`
+- `labels` - (Optional, map) User-defined labels (key-value pairs) should be created with.
 
 ## Attributes Reference
 
 The following attributes are exported:
 
-- `id` - Unique ID of the server.
-- `name` - Name of the server.
-- `server_type` - Name of the server type.
-- `image` - Name or ID of the image the server was created from.
-- `location` - The location name.
-- `datacenter` - The datacenter name.
-- `backup_window` - The backup window of the server, if enabled.
-- `iso` - Name of the mounted ISO image.
-- `ipv4_address` - The IPv4 address.
-- `ipv6_address` - The first IPv6 address of the assigned network.
-- `ipv6_network` - The IPv6 network.
-- `status` - The status of the server.
-- `labels` - User-defined labels (key-value pairs)
+- `id` - (int) Unique ID of the server.
+- `name` - (string) Name of the server.
+- `server_type` - (string) Name of the server type.
+- `image` - (string) Name or ID of the image the server was created from.
+- `location` - (string) The location name.
+- `datacenter` - (string) The datacenter name.
+- `backup_window` - (string) The backup window of the server, if enabled.
+- `iso` - (string) Name of the mounted ISO image.
+- `ipv4_address` - (string) The IPv4 address.
+- `ipv6_address` - (string) The first IPv6 address of the assigned network.
+- `ipv6_network` - (string) The IPv6 network.
+- `status` - (string) The status of the server.
+- `labels` - (map) User-defined labels (key-value pairs)
 
 ## Import
 

--- a/website/docs/r/ssh_key.html.md
+++ b/website/docs/r/ssh_key.html.md
@@ -24,18 +24,18 @@ resource "hcloud_ssh_key" "default" {
 
 The following arguments are supported:
 
-- `name` - (Required) Name of the SSH key.
-- `public_key` - (Required) The public key. If this is a file, it can be read using the file interpolation function
+- `name` - (Required, string) Name of the SSH key.
+- `public_key` - (Required, string) The public key. If this is a file, it can be read using the file interpolation function
 
 ## Attributes Reference
 
 The following attributes are exported:
 
-- `id` - The unique ID of the key.
-- `name` - The name of the SSH key
-- `public_key` - The text of the public key
-- `fingerprint` - The fingerprint of the SSH key
-- `labels` - User-defined labels (key-value pairs)
+- `id` - (int) The unique ID of the key.
+- `name` - (string) The name of the SSH key
+- `public_key` - (string) The text of the public key
+- `fingerprint` - (string) The fingerprint of the SSH key
+- `labels` - (map) User-defined labels (key-value pairs)
 
 ## Import
 

